microproxy/event/manager.py

```diff
@@ -0,0 +1,47 @@
+import zmq
+from zmq.eventloop import zmqstream
+
+import json
+
+from microproxy.utils import get_logger
+from replay import ReplayHandler
+
+logger = get_logger(__name__)
+
+
+class EventManager(object):
+    def __init__(self, config, proxy_server, handler=None, zmq_stream=None):
+        super(EventManager, self).__init__()
+        self.handler = handler or EventHandler(config, proxy_server)
+        self.zmq_stream = zmq_stream or self._create_stream(config)
+        self._start()
+
+    def _create_stream(self, config):  # pragma: no cover
+        context = zmq.Context()
+        socket = context.socket(zmq.PULL)
+        socket.bind(config["events_channel"])
+        logger.info("EventManager is listening at {0}".format(config["events_channel"]))
+        return zmqstream.ZMQStream(socket)
+
+    def _start(self):
+        self.zmq_stream.on_recv(self._on_recv)
+
+    def _on_recv(self, msg_parts):
+        message = msg_parts[0]
+        try:
+            event = json.loads(message)
+        except:
+            logger.error("Wrong message received: {0}".format(message))
+        else:
```

Comment:
The code you provide, will turn into that when parse a wrong json message,
there is no need to call self.handler.handle_event, but still call the function with event=None.
Is it the behavior you want?